OpenRA.Mods.Common/Traits/Render/Hovers.cs

```diff
@@ -43,7 +43,9 @@ public IEnumerable<IRenderable> ModifyRender(Actor self, WorldRenderer wr, IEnum
 			if (self.World.Paused || IsTraitDisabled)
 				return r;
 
-			var visualOffset = !aircraft || self.CenterPosition.Z > 0 ? (int)Math.Abs((self.ActorID + Game.LocalTick) / 5 % 4 - 1) - 1 : 0;
+			var visualOffset = !aircraft || self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length > 0
```

Comment:
This would have to be `var visualOffset = aircraft == null || self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length >= aircraft.MinAirborneAltitude`